src/main/java/com/emc/mongoose/storage/driver/pravega/io/DataItemTimestampSerializer.java

```diff
@@ -0,0 +1,68 @@
+package com.emc.mongoose.storage.driver.pravega.io;
+
+import com.emc.mongoose.base.item.DataItem;
+import com.emc.mongoose.base.logging.Loggers;
+import com.github.akurilov.commons.system.SizeInBytes;
+import io.pravega.client.stream.Serializer;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.ByteBuffer;
+
+public class DataItemTimestampSerializer<I extends DataItem>
+	implements Serializer<I>, Serializable {
+
+	private final boolean useDirectMem;
+
+	/**
+	 * @param useDirectMem Specifies whether should it use direct memory for the resulting buffer containing the
+	 *                     serialized data or not. Using the direct memory may lead to the better performance in case of
+	 *                     large data chunks but less safe.
+	 */
+	public DataItemTimestampSerializer(final boolean useDirectMem) {
```

Comment:
We don't know inside serializer if we should add timestamp or no, so a new serializer is added. I'd like to hear suggestions if there are any.

Reply:
just add one more flag arg

Reply:
that's how it works in this PR